server/repository/src/db_diesel/frontend_plugin_row.rs

```diff
@@ -0,0 +1,176 @@
+use super::{
+    frontend_plugin_row::frontend_plugin::dsl as frontend_plugin_dsl, ChangeLogInsertRow,
+    ChangelogRepository, ChangelogTableName, RowActionType, StorageConnection,
+};
+
+use crate::{repository_error::RepositoryError, Delete, Upsert};
+use diesel::prelude::*;
+use diesel_derive_enum::DbEnum;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub struct FrontendPluginFile {
+    pub file_name: String,
+    pub file_content_base64: String,
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize)]
+pub struct FrontendPluginFiles(pub Vec<FrontendPluginFile>);
+
+impl From<String> for FrontendPluginFiles {
```

Comment:
This sort of serialization was first introduced in backed plugins, it's quite convenient, can be made generic